boards/arm/mimxrt1010_evk/mimxrt1010_evk.dts

```diff
@@ -46,7 +46,7 @@ arduino_serial: &lpuart1 {};
 &flexspi {
 	reg = <0x402a8000 0x4000>, <0x60000000 DT_SIZE_M(16)>;
 	at25sf128a: at25sf128a@0 {
-		compatible = "adesto,at25sf128a", "jedec,spi-nor";
+		compatible = "nxp,imx-flexspi-nor";
```

Comment:
It'd be fine to have this be:

    compatible = "adesto,at25sf128a", "nxp,imx-flexspi-nor";

Then if somebody has an Adesto-specific driver it'd be picked up.  It also makes more clear exactly what the flash chip is.  Not required, but suggested.

The same applies to all the other flash device nodes.

Reply:
I like how this makes the flash chip more clear, but for it to work I think we'd need a compatible something like `nxp,imx-flexspi-nor-at25sf128a`. Because an Adesto-specific driver would be different for each of the various SoC QSPI peripherals.

Reply:
I'm not sure that's necessarily true.  I'm thinking this would be useful in somebody's out-of-tree application where they didn't care that the underlying peripheral isn't represented in the compatible name, but their needs are such that their dedicated custom driver is preferred over the stock Zephyr one.

The primary purpose of the compatible is to accurately describe the device that's present in hardware, not the driver implementation to talk to that device.  Kconfig figures out the best device.  QSPI bus devices are perhaps an extreme case of this, but could be supported if we used `jedec,qspi-nor` and have the vendor-specific drivers all support that compatible, if it weren't for the fact the vendor drivers tend to need vendor-specific devicetree properties.